Conformity/custom-rule/custom-rule.py

```diff
@@ -0,0 +1,269 @@
+
+	response = requests.get(url=endpoint, headers=header)
+	rule = response.json()
+	ruleId = rule["data"][0]["id"]
+	ruleFilename = f'{ruleId}.yaml'
+
+	with open(f'{workspaceFolder}/{ruleFilename}', 'w') as configWriter:
+		yaml.dump(data=rule["data"][0], stream=configWriter, indent=4, sort_keys=False)
+
+	print(f'Rule saved to file {ruleFilename}')
+	pass
+
+def update(args):
+	header, region = readConfiguration()
+	updatedRule = {}
+
+	with open(f'{workspaceFolder}/{args.file}', 'r') as ruleReader:
+		rule = yaml.safe_load(stream=ruleReader)
+
+		endpoint = f'{apiEndpoint.format(region)}/{rule.get("id")}' 
+		
+		response = requests.put(url=endpoint, json=rule["attributes"], headers=header)
+		updatedRule = response.json()
+	
+	ruleId = updatedRule["data"]["id"]
+	ruleFilename = f'{ruleId}.yaml'
+
+	with open(f'{workspaceFolder}/{ruleFilename}', 'w') as configWriter:
+		yaml.dump(data=updatedRule["data"], stream=configWriter, indent=4, sort_keys=False)
+
+	print(f'Rule saved to file {ruleFilename}')
+
+	pass
+
+def delete(args):
+	header, region = readConfiguration()
+	endpoint = f'{apiEndpoint.format(region)}/{args.ruleId}'
+	
+	response = requests.delete(url=endpoint, headers=header)
+
+	print(f'Rule deleted. Run "custom-rule list" for updated list of rules')
+	pass
+
+def list(args):
+	header, region = readConfiguration()
+	endpoint = apiEndpoint.format(region)
+
+	response = requests.get(url=endpoint, headers=header)
+	rules = response.json()
+
+	with open(f'{workspaceFolder}/rules.yaml', 'w') as configWriter:
+		yaml.dump(data=rules["data"], stream=configWriter, indent=4, sort_keys=False)
+	
+	print("Rules were stored in rule.yaml file.")
+	pass
+
+def run(args):
+	header, region = readConfiguration()
+	endpoint = f'{apiEndpoint.format(region)}/run?accountId={args.accountId}&resourceData=true'
+	
+	rule = {}
+	with open(f'{workspaceFolder}/{args.ruleId}.yaml', 'r') as ruleReader:
+		rule = {
+			"configuration": yaml.safe_load(stream=ruleReader)["attributes"]
+		}
+	
+	rule["configuration"]["resourceId"] = args.resourceId
+	response = requests.post(url=endpoint, json=rule, headers=header)
+
+	dryrunFilename = f'{args.ruleId}.run.yaml'
+	with open(f'{workspaceFolder}/{dryrunFilename}', 'w') as configWriter:
+		yaml.dump(data=response.json(), stream=configWriter, indent=4, sort_keys=False)
+
+	print(f'Run trace saved to file {dryrunFilename}. Note that resourceData information can be used for rule development')
+	pass
+
+def showProviders(args):
+	response = requests.get(url="https://us-west-2.cloudconformity.com/v1/providers")
+	providers = response.json()
+	
+	print(json.dumps([x.get("id") for x in providers["data"]], indent=4))
+	pass
+
+def showServices(args):
+	response = requests.get(url="https://us-west-2.cloudconformity.com/v1/services")
+	services = response.json()
+	
+	print(json.dumps([ x["id"] for x in services["data"] if x["attributes"]["provider"] == args.provider ], indent=4))
+	pass
+
+def showResourceTypes(args):
```

Comment:
Here when you ask for a inexistent service or when you do a typo, instead of show a empty array, you could print a message to instruct the user that is a inexistent service or a typo in the name, just for better clarity, like:

```
python custom-rule.py show-resource-types --service 12345
[]
```
You could print this way:

```
python custom-rule.py show-resource-types --service 12345

Inexistent service, check the name and try again
```

Reply:
Some services don't have rules yet so they will not show when querying for services.